src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,148 @@
+    private static final Integer INVALID_LARGE_AGE = 130;
+    private static final String VALID_LOGIN = "VALID_LOGIN";
+    private static RegistrationServiceImpl registrationService;
+    private static User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setLogin(VALID_LOGIN);
+        user.setPassword(VALID_PASSWORD);
+        user.setAge(VALID_AGE);
+    }
+
+    @Test
+    void register_validUser_ok() {
+        registrationService.register(user);
+        assertTrue(Storage.people.contains(user),
+                "Registration valid user should add user to Storage.");
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        assertThrows(UserDataException.class,
+                () -> registrationService.register(null),
+                "Registration NULL user should throw IncorrectUserDataException.");
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        user.setLogin(null);
+        assertThrows(UserDataException.class,
+                () -> registrationService.register(user),
+                "Registration user with NULL login"
+                        + " should throw IncorrectUserDataException.");
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        user.setAge(null);
+        assertThrows(UserDataException.class,
+                () -> registrationService.register(user),
+                "Registration user with NULL age"
+                        + " should should throw IncorrectUserDataException.");
+    }
+
+    @Test
+    void register_negativeAge_notOk() {
+        user.setAge(NEGATIVE_INVALID_AGE);
+        assertThrows(UserDataException.class,
+                () -> registrationService.register(user),
+                "Registration user with age < 0"
+                        + " should should throw IncorrectUserDataException.");
+    }
+
+    @Test
+    void register_ageGreaterThenMaxAge_notOk() {
+        user.setAge(INVALID_LARGE_AGE);
+        assertThrows(UserDataException.class,
+                () -> registrationService.register(user),
+                "Registration user with age > 110"
+                        + " should should throw IncorrectUserDataException.");
+    }
+
+    @Test
+    void register_ageLessThenEighteen_notOk() {
+        user.setAge(MIN_INVALID_AGE);
+        assertThrows(UserDataException.class,
+                () -> registrationService.register(user),
+                "Registration user with age < 18"
+                        + " should throw IncorrectUserDataException.");
+    }
+
+    @Test
+    void register_ageValidEighteen_ok() {
+        user.setAge(MIN_VALID_AGE);
+        registrationService.register(user);
+        assertTrue(Storage.people.contains(user),
+                "Registration user with age = 18 should add user to Storage.");
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
```

Comment:
you should move this method closer to all other null tests, but its not critical